### PR TITLE
EWPP-472: Allow remote files in Document media bundle.

### PR DESCRIPTION
**Pull request by 22Alexandra**
Out of the box, the Document media bundle provides the capability to upload a local document or choose a remote URL for the document, whose metadata will be pulled from that remote file (such as file type and size).

If you are updating from a version before `1.10.0` and you would like to benefit from the remote document capability, all you have to do is:

* Run the database updates as you normally would
* Edit the Document media form display and place the `File Type` and `Remote File` fields below the `Name` field. The most common order would be:
  * Name
  * File Type
  * File
  * Remote File

This will ensure the functionality kicks in.

### CODE REVIEW

File: src/Plugin/Validation/Constraint/DocumentMediaConstraint.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * Message shown when the remote file is missing.
+   *
+   * @var string
+   */
+  public $messageMissingRemoteFile = 'The document is configured to be remote, please reference a remote file.';
+
+  /**
+   * Message shown when the remote file is missing.
```
Comment (22Alexandra) [inline src/Plugin/Validation/Constraint/DocumentMediaConstraint.php:35]:
```suggestion
   * Message shown when the file type is missing.
```

File: oe_media.post_update.php [hunk 2]
```diff
@@ -30,3 +32,115 @@ function oe_media_post_update_document_private_files(): void {
...
   }
 }
+
+/**
+ * Add new fields to document media bundle to allow remote files.
+ */
+function oe_media_post_update_00001_remote_file(array &$sandbox) {
+  // Enable file_link module.
+  $module_installer = \Drupal::service('module_installer');
+  $module_installer->install(['options', 'file_link']);
```
Comment (22Alexandra) [inline oe_media.post_update.php:42]:
```suggestion
  \Drupal::service('module_installer')->install(['options', 'file_link']);
```

File: tests/Kernel/DocumentMediaTest.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+use Drupal\file\Entity\File;
+use Drupal\media\MediaInterface;
+
+/**
+ * Tests the document media type.
+ */
+class DocumentMediaTest extends MediaTestBase {
+
+  /**
+   * Tests that the the correct field keeps the value depending on the type.
```
Comment (22Alexandra) [inline tests/Kernel/DocumentMediaTest.php:16]:
```suggestion
   * Tests that the correct field keeps the value depending on the type.
```

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

File: tests/Functional/DocumentMediaTest.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    $this->drupalLogin($this->drupalCreateUser([], '', TRUE));
+  }
+
+  /**
+   * Tests the remote file document media.
+   */
+  public function testRemoteFile(): void {
+    $this->drupalGet('media/add/document');
+
+    $assert_session = $this->assertSession();
```
[the remark this answers is not in the document]
Comment (22Alexandra) [reply tests/Functional/DocumentMediaTest.php:49]: The behat scenario. here is not enough? https://github.com/openeuropa/oe_media/pull/131/files#diff-0df9a816419d7a9cceb781c2310ecdc2b6da3352073332a410769098589408d4R58-R71

[the remark this answers is not in the document]
Comment (upchuk) [reply tests/Functional/DocumentMediaTest.php:49]: Yes, it's covered in Behat, both for regular and IEF embed no?

**Review by imanoleguskiza — DISMISSED** [review]

**Review by upchuk — APPROVED** [review]

**Review by hernani — APPROVED** [review]

**Pull request merged**